app/cdap/components/SourceControlManagement/LocalPipelineListView/index.tsx

```diff
@@ -0,0 +1,163 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import PrimaryContainedButton from 'components/shared/Buttons/PrimaryContainedButton';
+import React, { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import {
+  countPushFailedPipelines,
+  getNamespacePipelineList,
+  pushSelectedPipelines,
+  reset,
+  resetPushStatus,
+  setLoadingMessage,
+  setLocalPipelines,
+  setNameFilter,
+  toggleCommitModal,
+  toggleShowFailedOnly,
+} from '../store/ActionCreator';
+import { SearchBox } from '../SearchBox';
+import LoadingSVGCentered from 'components/shared/LoadingSVGCentered';
+import T from 'i18n-react';
+import { CommitModal } from './CommitModal';
+import cloneDeep from 'lodash/cloneDeep';
+import styled from 'styled-components';
+import PrimaryTextButton from 'components/shared/Buttons/PrimaryTextButton';
+import { LocalPipelineTable } from './PipelineTable';
+import { useOnUnmount } from 'services/react/customHooks/useOnUnmount';
+
+const PREFIX = 'features.SourceControlManagement.push';
+
+const StyledSelectionStatusDiv = styled.div`
+  display: flex;
+  text-align: center;
+  margin: 10px 2px;
+
+  div {
+    margin-right: 10px;
+    padding-top: 4px;
+  }
+`;
+
+export const LocalPipelineListView = () => {
+  const {
+    ready,
+    localPipelines,
+    nameFilter,
+    selectedPipelines,
+    commitModalOpen,
+    loadingMessage,
+    showFailedOnly,
+  } = useSelector(({ push }) => push);
+
+  const pushFailedCount = countPushFailedPipelines();
+
+  useEffect(() => {
+    if (!ready) {
+      getNamespacePipelineList(getCurrentNamespace(), nameFilter);
+    }
+  }, [ready]);
+
+  useOnUnmount(() => reset());
+
+  const onPushSubmit = (commitMessage: string) => {
+    resetPushStatus();
+    const pushedPipelines = cloneDeep(localPipelines);
+    const payload = {
+      commitMessage,
+    };
+    toggleCommitModal();
+    pushSelectedPipelines(
+      getCurrentNamespace(),
+      selectedPipelines,
+      payload,
+      setLoadingMessage
+    ).subscribe({
+      next(res: any) {
+        const currentPipeline = pushedPipelines.find((pipeline) => pipeline.name === res.name);
+        if (res.statusCode !== 200) {
+          currentPipeline.error = res.message;
+          currentPipeline.success = false;
+        } else {
+          currentPipeline.success = true;
+        }
+      },
+      complete() {
+        setLocalPipelines(pushedPipelines);
+        setLoadingMessage(null);
+      },
+    });
+  };
+
+  return (
+    <div style={{ height: 'calc(100% - 150px)' }}>
```

Comment:
This should be like SelectedPipelinesContainer or something instead of this div style=